apps/client/src/pages/CheckerPage.js

```diff
@@ -0,0 +1,231 @@
+import React, { useContext } from "react";
+import { Helmet } from "react-helmet";
+import { Redirect } from "react-router-dom";
+
+import Conclusion from "../components/Conclusion";
+import DebugDecisionTable from "../components/DebugDecisionTable";
+import Layout from "../components/Layouts/DefaultLayout";
+import LocationInput from "../components/Location/LocationInput";
+import LocationResult from "../components/Location/LocationResult";
+import PrintDetails from "../components/PrintDetails";
+import Questions from "../components/Questions";
+import {
+  StepByStepItem,
+  StepByStepNavigation,
+} from "../components/StepByStepNavigation";
+import { SessionContext } from "../context";
+import withChecker from "../hoc/withChecker";
+import { geturl, routes } from "../routes";
+
+const CheckerPage = ({ checker, topic, resetChecker }) => {
+  const sessionContext = useContext(SessionContext);
+  const { slug, sttrFile, text } = topic;
+  // OLO Flow does not have questionIndex
+  const { questionIndex } = sttrFile ? sessionContext[topic.slug] : 0;
+
+  //@TODO: We shoudn't need this redirect. We need to refactor this
+  if (!sessionContext[slug]) {
+    return <Redirect to={geturl(routes.intro, topic)} />;
+  }
+
+  const { activeComponents, answers, finishedComponents } = sessionContext[
+    slug
+  ];
+
+  // Only one component can be active at the same time.
+  const setActiveState = (component) => {
+    sessionContext.setSessionData([slug, { activeComponents: [component] }]);
+  };
+
+  /**
+   * Set given components to the given finished state
+   *
+   * @param { Object[] | string } component - name or names of the components
+   * @param { boolean } value - Set components to finished state or remove from finished state.
+   */
+  const setFinishedState = (component, value) => {
+    // If component is only a string, we make it a array first
+    const allComponents = Array.isArray(component) ? component : [component];
+
+    // If value is false, remove the components from the fishedComponents array
+    const newFinishedComponents =
+      typeof value === "boolean" && value === false
+        ? finishedComponents?.filter((c) => !allComponents.includes(c)) || []
+        : [...finishedComponents, ...allComponents];
+
+    // Save the new array to the context
+    sessionContext.setSessionData([
+      slug,
+      { finishedComponents: newFinishedComponents },
+    ]);
+  };
+
+  const isActive = (component, finished = false) => {
```

Comment:
`finished` prop seems weird here. Either rename or move the logic to `isFinished` ?